tests/dofs/dof_tools_22a.cc

```diff
@@ -0,0 +1,160 @@
+#include <set>
+#include <cstdio>
+
+
+void test ()
+{
+
+    unsigned int np = Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD);
+    unsigned int myid = Utilities::MPI::this_mpi_process (MPI_COMM_WORLD);
+    const int dim = 2;
+    //setup system
+    dealii::parallel::distributed::Triangulation<dim> triangulation(MPI_COMM_WORLD);
+    
+    GridGenerator::hyper_rectangle (triangulation,
+                                    Point<dim>(0,0),
+                                    Point<dim>(1,1),
+                                    true);
+
+    triangulation.refine_global(1);
+    //extra refinement to generate hanging nodes
+        for (typename Triangulation<dim>::active_cell_iterator
+             cell = triangulation.begin_active();
+             cell != triangulation.end(); ++cell)
+                if (cell->center()(0) > 0.49)
+                    cell->set_refine_flag ();
+    
+    triangulation.prepare_coarsening_and_refinement();
+    triangulation.execute_coarsening_and_refinement ();
+    triangulation.repartition();
+    
+    const FESystem<dim> fe_system (FE_Q<dim>(2),1,
+                                   FE_DGQ<dim>(2),1);
+    
+    DoFHandler<dim> dh (triangulation);
+
+    dh.distribute_dofs (fe_system);
+
+    //couple the internal DoFs of both finite elements. Only couple the face terms of the discontinuous element.
+    Table<2,DoFTools::Coupling> coupling (2, 2);
+    Table<2,DoFTools::Coupling> flux_coupling (2, 2);
+    
+    for (unsigned int i=0; i<2; ++i)
+        for (unsigned int j=0; j<2; ++j)
+        {
+            coupling[i][j] = DoFTools::none;
+            flux_coupling[i][j] = DoFTools::none;
+        }
+            
+    coupling[0][0] = DoFTools::always;
+    coupling[1][1] = DoFTools::always;
+    flux_coupling[1][1] = DoFTools::always;
+    
+    IndexSet relevant_partitioning (dh.n_dofs());
+    DoFTools::extract_locally_relevant_dofs (dh, relevant_partitioning);
+
+    //generate hanging node constraints
+    ConstraintMatrix constraints;
+    constraints.reinit(relevant_partitioning);
+    DoFTools::make_hanging_node_constraints (dh,
+                                             constraints);
+    constraints.close();
+    
+    //generate sparsity pattern
+    TrilinosWrappers::SparsityPattern sp (dh.locally_owned_dofs(),
+                                          dh.locally_owned_dofs(),
+                                          relevant_partitioning,
+                                          MPI_COMM_WORLD);
+    
+    DoFTools::make_flux_sparsity_pattern (dh, sp,
+                                          constraints,true,
+//                                          coupling,
+//                                          flux_coupling,
+                                          Utilities::MPI::this_mpi_process(MPI_COMM_WORLD));
+    
+    sp.compress();
+    //output
+    
+    std::ofstream logfile("output");
+    deallog.attach(logfile);
+    deallog.threshold_double(1.e-10);
```

Comment:
In particular, I don't think you need it because you only write into this file from processor 0, and that processor has already opened it in `main()`.